### PR TITLE
Loan pallet fixes

### PR DESCRIPTION
**Pull request by vedhavyas**
This PR:
- Moves from Millisecs to seconds to be consistent with pool pallet
- Update tests to set milliseconds since timestamp pallets expects the Milliseconds in each block
- Ensure NAV is set to default when pool is initialised
- Deal with rounding error when trying to repay in the same block as borrow.
	- **I leaned towards allowing repay starting from the next block. any concerns ?**

### CODE REVIEW

File: pallets/loan/src/lib.rs [hunk 1]
```diff
@@ -240,8 +239,8 @@ pub mod pallet {
 		/// Emits when operation is done on an inactive loan
 		ErrLoanNotActive,
 
-		/// Emits when epoch time is overflowed
-		ErrEpochTimeOverflow,
+		// Emits when borrow and repay doesn't happen in the same block
```
Comment (offerijns) [inline pallets/loan/src/lib.rs:242]:
```suggestion
		// Emits when borrow and repay happens in the same block
```

**Comment by mustermeiszer** [comment]
I think lending and repaying in one block should not have a real use-case besides gaming the system in a unwanted way, right? So I think leaning to delaying it to the next block seems reasonable.

File: pallets/loan/src/functions.rs [hunk 2]
```diff
@@ -330,6 +330,16 @@ impl<T: Config> Pallet<T> {
...
+				let now: u64 = Self::time_now();
+
+				// ensure current time is more than origination time
+				// this is mainly to deal with how we calculate debt while trying to repay
+				// therefore we do not let users repay at same instant origination happened
+				ensure!(
+					now > loan_info.origination_date,
+					Error::<T>::ErrRepayTooEarly
+				);
+
```
Comment (mustermeiszer) [inline pallets/loan/src/functions.rs:342]: Can `Self::time_now()` differ in one block?

**Review by mustermeiszer — APPROVED** [review]
LGTM! 

Just a question, regarding if `T::Time::now()` can differ in a block.

File: pallets/loan/src/functions.rs (hunk 2 above)
Comment (mustermeiszer) [reply pallets/loan/src/functions.rs:342]: Nevermind, its set on a per block basis.

**Pull request merged**